tests/test_authz_authorizer.py

```diff
@@ -38,37 +32,161 @@ def test__init__(self):
+class TestAuthorizerSetupMethod:
+    @staticmethod
+    def _mocked_make_authorizer(token_payload: dict) -> Authorizer:
+        with patch("lbz.authz.authorizer.decode_jwt", lambda _: token_payload):
+            return Authorizer("xx", "test_resource", "permission_name")
+
+    @pytest.fixture(autouse=True)
+    def setup_method(self, base_auth_payload) -> None:
+        # pylint: disable=attribute-defined-outside-init
+        self.token_payload = base_auth_payload
+        self.authz = self._mocked_make_authorizer(self.token_payload)
+
+    def test_wrong_jwt_authz_payload_raises_permission_denied(self) -> None:
+        with pytest.raises(PermissionDenied):
+            self._mocked_make_authorizer({})
+
+    def test_check_deny_res(self) -> None:
+        self.authz.deny = {"test_resource": ALL}
+        with pytest.raises(PermissionDenied):
+            self.authz._check_deny()  # pylint: disable=protected-access
+        assert self.authz.outcome == DENY
+
+    def test_check_allow_all(self) -> None:
+        self.authz._check_allow_and_set_resources()  # pylint: disable=protected-access
+        assert self.authz.outcome == ALLOW
+
+    def test_check_deny_any(self) -> None:
+        self.authz.deny = {ALL: ALL}
+        with pytest.raises(PermissionDenied):
+            self.authz._check_deny()  # pylint: disable=protected-access
+        assert self.authz.outcome == DENY
+
+    def test_check_access_check_deny(self) -> None:
+        authz = self._mocked_make_authorizer({**self.token_payload, "deny": {ALL: ALL}})
+        with pytest.raises(PermissionDenied):
+            authz.check_access()
+
+    def test_check_access_outcome_deny(self) -> None:
+        authz = self._mocked_make_authorizer({**self.token_payload, "allow": {ALL: None}})
+        with pytest.raises(PermissionDenied):
+            authz.check_access()
+
+    def test__check_resource(self) -> None:
+        with patch.object(self.authz, "_deny_if_all", new_callable=MagicMock()) as mocked_deny:
+            self.authz._check_resource({"res": "xxx"})  # pylint: disable=protected-access
+            mocked_deny.assert_has_calls(
+                [
+                    call({"res": "xxx"}),
+                    call("res"),
+                    call("xxx"),
+                ]
+            )
+
+    def test_validate_allow_all(self) -> None:
+        self.authz.check_access()
+        assert self.authz.outcome == ALLOW
+        assert self.authz.restrictions == {"allow": "*", "deny": None}
+
+    def test_check_allow_fail_all(self) -> None:
+        self.authz.allow = None
+        with pytest.raises(PermissionDenied):
+            self.authz._check_allow_and_set_resources()  # pylint: disable=protected-access
+            assert self.authz.outcome == ALLOW
+
+    def test_check_allow_domain(self) -> None:
+        self.authz.allow = {"test_resource": ALL}
+        self.authz._check_allow_and_set_resources()  # pylint: disable=protected-access
```

Comment:
Why do you test a protected method? That can be used differently inside the class. I think testing `check_access()` makes the most sense.

Reply:
becouse here we are testing the authorize module ergo all its private and magick methods.

Reply:
But I am speaking about the general approach to testing - if something has been marked as protected, you shouldn't use it outside the class/module. If you need to do this, that starts smelling like a problem either in the test approach or the code in general 😬 

Of course, you can simply mark that as public (not recommended) or test something that is public and what will be really used by the outside world 😉

Reply:
agreed to disagree offline.

Reply:
As below: we will come back to this topic in the future, because it is an important thing when designing the code 😉